internal/transform/operation.go

```diff
@@ -0,0 +1,345 @@
+	case "buying":
+		operationDetails.BuyingAssetType = assetType
+		if asset.Type != xdr.AssetTypeAssetTypeNative {
+			operationDetails.BuyingAssetIssuer = issuer
+			operationDetails.BuyingAssetCode = code
+		}
+
+	case "selling":
+		operationDetails.SellingAssetType = assetType
+		if asset.Type != xdr.AssetTypeAssetTypeNative {
+			operationDetails.SellingAssetIssuer = issuer
+			operationDetails.SellingAssetCode = code
+		}
+
+	case "source":
+		operationDetails.SourceAssetType = assetType
+		if asset.Type != xdr.AssetTypeAssetTypeNative {
+			operationDetails.SourceAssetIssuer = issuer
+			operationDetails.SourceAssetCode = code
+		}
+
+	default:
+		operationDetails.AssetType = assetType
+		if asset.Type != xdr.AssetTypeAssetTypeNative {
+			operationDetails.AssetIssuer = issuer
+			operationDetails.AssetCode = code
+		}
+
+	}
+	return nil
+}
+func convertPathToAssetOutput(initialPath []xdr.Asset) []AssetOutput {
+	if len(initialPath) == 0 {
+		return nil
+	}
+	var path = []AssetOutput{}
+	for _, pathAsset := range initialPath {
+		var assetType, code, issuer string
+		err := pathAsset.Extract(&assetType, &code, &issuer)
+		if err != nil {
+			return nil
+		}
+
+		path = append(path, AssetOutput{
+			AssetType:   assetType,
+			AssetIssuer: issuer,
+			AssetCode:   code,
+		})
+	}
+	return path
+}
+func addOperationFlagToOperationDetails(operationDetails *Details, flag int32, prefix string) {
+	var intFlags []int32
+	var stringFlags []string
+
+	if (flag & int32(xdr.AccountFlagsAuthRequiredFlag)) > 0 {
+		intFlags = append(intFlags, int32(xdr.AccountFlagsAuthRequiredFlag))
+		stringFlags = append(stringFlags, "auth_required")
+	}
+
+	if (flag & int32(xdr.AccountFlagsAuthRevocableFlag)) > 0 {
+		intFlags = append(intFlags, int32(xdr.AccountFlagsAuthRevocableFlag))
+		stringFlags = append(stringFlags, "auth_revocable")
+	}
+
+	if (flag & int32(xdr.AccountFlagsAuthImmutableFlag)) > 0 {
+		intFlags = append(intFlags, int32(xdr.AccountFlagsAuthImmutableFlag))
+		stringFlags = append(stringFlags, "auth_immutable")
+	}
+
+	switch prefix {
+	case "set":
+		operationDetails.SetFlags = intFlags
+		operationDetails.SetFlagsString = stringFlags
+
+	case "clear":
+		operationDetails.ClearFlags = intFlags
+		operationDetails.ClearFlagsString = stringFlags
+	}
+}
+
+func extractOperationDetails(operation xdr.Operation, transaction ingestio.LedgerTransaction, operationIndex int32) (Details, error) {
+	outputDetails := Details{}
+	sourceAccount := getOperationSourceAccount(operation, transaction)
+	sourceAccountAddress, _ := utils.GetAccountAddressFromMuxedAccount(sourceAccount)
+	operationType := operation.Body.Type
+	allOperationResults, ok := transaction.Result.OperationResults()
+	if !ok {
+		return Details{}, fmt.Errorf("Could not access any results for this transaction")
+	}
+
+	currentOperationResult := allOperationResults[operationIndex]
+	switch operationType {
+	case xdr.OperationTypeCreateAccount:
+		op := operation.Body.MustCreateAccountOp()
```

Comment:
I'm wary of the `Must` here and below. Seems like it might panic rather than returning an error message, which would help explain where the user should re-ingest.